apps/web/app/[language]/homestead/materials/page.tsx

```diff
@@ -0,0 +1,194 @@
+import { Headline } from '@gw2treasures/ui/components/Headline/Headline';
+import {
+  createDataTable,
+  type DataTableColumnProps,
+  type DataTableColumnSelectionProps,
+  type DataTableDynamicColumnsProps,
+  type DataTableProps,
+} from '@gw2treasures/ui/components/Table/DataTable';
+import type { FC } from 'react';
+import {
+  fiber,
+  FIBER_ID,
+  metal,
+  METAL_ID,
+  wood,
+  WOOD_ID,
+  type ConversionRate,
+  type RefinedSources,
+} from './data';
+import { Switch } from '@gw2treasures/ui/components/Form/Switch';
+import type { PageProps } from '@/lib/next';
+
+const getItems = cache(
+  async (ids: number[]) => {
+    const items = await db.item.findMany({
+      where: { id: { in: ids }},
+      select: {
+        ...linkProperties,
+        tpTradeable: true,
+        tpCheckedAt: true,
+        buyPrice: true,
+        sellPrice: true,
+      },
+    });
+
+    return Object.fromEntries(groupById(items).entries());
+  },
+  ['homestead-refined-mats'],
+  { revalidate: 60 * 5 }
+);
+
+export default async function RefinedMaterialsPage({ searchParams: { efficiency: rawEfficiency }}: PageProps) {
+  const allIds = [
+    ...Object.keys(wood),
+    ...Object.keys(metal),
+    ...Object.keys(fiber),
+    FIBER_ID,
+    WOOD_ID,
+    METAL_ID,
+  ].map((v) => Number(v));
+
+  // get items from db that unlock some node
+  const items = await getItems(allIds);
+  const efficiency = Math.max(1, Math.min(3, Number(rawEfficiency) || 3));
+
+  const getRefinedData = (source: RefinedSources, id: number) => {
+    return {
+      item: items[id],
+      sources: Object.entries(source).map(([sourceId, costs]) => ({
+        id: sourceId,
+        item: items[sourceId],
+        rate: costs[efficiency - 1],
+      })).filter((v) => v.item),
+    };
+  };
+
+  // add the item to each node
+  const metalData = getRefinedData(metal, METAL_ID);
+  const MetalTable = createDataTable(metalData.sources, ({ id }) => id);
+  
+  const woodData = getRefinedData(wood, WOOD_ID);
+  const WoodTable = createDataTable(woodData.sources, ({ id }) => id);
+  
+  const fiberData = getRefinedData(fiber, FIBER_ID);
+  const FiberTable = createDataTable(fiberData.sources, ({ id }) => id);
+
+  return (
+    <>
+      <Description>
+        <Trans id="homestead.materials.description"/>
+      </Description>
+      <Description>
+        <Trans id="homestead.materials.help"/>
+      </Description>
+
+      <label style={{ display: 'flex', gap: 16, alignItems: 'center' }}>
+        <Trans id="homestead.materials.efficiency"/>:
+        <Switch>
+          <Switch.Control type="link" replace active={efficiency === 1} href="/homestead/materials?efficiency=1">1</Switch.Control>
+          <Switch.Control type="link" replace active={efficiency === 2} href="/homestead/materials?efficiency=2">2</Switch.Control>
+          <Switch.Control type="link" replace active={efficiency === 3} href="/homestead/materials">3</Switch.Control>
+        </Switch>
+      </label>
```

Comment:
I didn't want to overcomplicate things by adding 3 different query params 😅 but yep, it might make sense.

I'm thinking if it should be possible to encode all of these into a single numeric value... easiest encoding would be 102 = metal=1, wood=0, fiber=2.